### PR TITLE
RCNN 논문 리뷰

### PR DESCRIPTION
**Pull request by cstack1994**
RCNN 관련 내용으로 정리해보았습니다.
논문을 세부적으로 다 해석하기에 어려움이 있어 블로그 및 강의를 
참고하여 이해할 수 있는 선까지 작성해보았습니다

### CODE REVIEW

File: docs/Computer Vision/Object Detection/R-CNN.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+- **장점**
+    - 동시대의 타 알고리즘 대비 높은 검출 정확도
+- **단점**
+    - 너무 느린 검출 시간과 복잡한 아키텍처 및 학습 프로세스
+    - 하나의 이미지마다 Selective Search를 수행하여 2,000개의 region 영역 이미지 도출
+    - 개별 이미지별 2,000개씩 생성된 regino  이미지를 CNN Feature map 생성
+    - Selective search, CNN Feature extractor, SVM, Bounding box regressor로 구성되어 있는 프로세스를 거쳐 학습 및 객체 검출이 되어야함
+    - 1장의 이미지를 object detection 하는데 약 50초 소요
+
```
Comment (KEJdev) [inline docs/Computer Vision/Object Detection/R-CNN.md:137]: 장점에 비해 단점이 많은데 R-CNN 논문이 왜 유명한지에 대해 추가적으로 적어주시면 좋을 것 같습니다.

답을 미리 알려드리자면, R-CNN논문이 유명한 이유는 2-stage detector들의 구조에 막대한 영향을 미쳤고, Obejct Detection에 CNN을 적용시켰다는 점이 의미가 있습니다.

File: docs/Computer Vision/Object Detection/R-CNN.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+---
+
+- 학습 과정
+    - Selective Search 알고리즘을 사용하여 이미지에서 객체 후보 영역 식별 후, 각 영역을 AlexNet CNN 아키텍처가 요구하는 고정 크기 (227x227)로 변환하여 CNN 입력을 통해 feature를 추출하고, feature를 기반으로 객체의 클래스를 분류하는 분류기 학습
+    - 객체 클래스 분류하는 분류기는 support vector machine(SVM)
+- 객체 검출
+    - 테스트 단계에서, 학습된 CNN을 사용하여 테스트 이미지 후보 영역에서 특징 추출하고, 학습된 분류기를 통해 각 영역의 객체 클래스 예측.
+ 
+<details>
+<summary>Classification</summary>
```
Comment (KEJdev) [inline docs/Computer Vision/Object Detection/R-CNN.md:96]: 학습과정에서 classifier로 sortmax를 쓰지 않고 SVM을 사용한 이유는 무엇인가요?

File: docs/Computer Vision/Object Detection/R-CNN.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+</aside>
+
+## RCNN Paper
+
+---
+
+- **Title :** Region-based Convolutional Networks for Accurate Object Detection and Segmentation
+- **Author :** Ross Girshick, Jeff Donahue, Trevor Darrell, Jitendra Malik
+- **Date of Publication :** 25 May 2015
+
```
Comment (KEJdev) [inline docs/Computer Vision/Object Detection/R-CNN.md:17]: 해당 논문을 왜 선정하게 되었는지에 대한 이유가 추가로 있으면 좋을 것 같습니다.

**Review by KEJdev — COMMENTED** [review]
민찬님~ 너무 수고 많으셨습니다!

File: docs/Computer Vision/Object Detection/R-CNN.md (hunk 1 above)
Comment (cstack1994) [reply docs/Computer Vision/Object Detection/R-CNN.md:137]: 피드백 감사합니답!! 단점의 내용이 다 같은 말인데 단점만 세부적으로 작성한것 같습니다.. ㅎ
말씀해주신 답변에 추가적으로 생각해봤을 때 아래 내용과 같은 이유일 것으로 판단됩니다!
1. 전통적인 머신러닝 한계를 극복을 위한 CNN 적용
2. 부족한 데이터 셋의 한계 극복을 위해 ImageNet pre-trained 모델 fine-tuning 기법 적용
3. 속도와 효율성 개선을 위해 기존 Sliding window 기법에서 Selective search 기법으로 전환
4. 객체 탐지 정밀도 향상을 위해 CNN 및 Bounding box regression 적용

**Comment by ethanj-dev** [comment]
민찬님 논문 리뷰하시느라 고생 많으셨습니다!!

File: docs/Computer Vision/Object Detection/R-CNN.md (hunk 2 above)
Comment (cstack1994) [reply docs/Computer Vision/Object Detection/R-CNN.md:96]: Softmax 함수는 다중 클래스 분류 문제에 주로 사용되며, 클래스 간의 확률 분포를 출력합니다. 하지만, RCNN에서는 대량의 배경 영역이 존재하고, 이러한 배경 영역들은 대부분의 클래스에 속하지 않으므로 다중클래스 분류인 Softmax 보다 이진 클래스 분류인 SVM을 사용하면, 각 클래스에 대해 독립적으로 이진 분류기를 훈련시킬 수 있으며, 이는 배경 클래스를 더 잘 처리할 수 있다는 결론과 SVM을 통한 실험 결과의 성능이 높았기 때문에 채택이 되었다고 합니다!

File: docs/Computer Vision/Object Detection/R-CNN.md (hunk 3 above)
Comment (cstack1994) [reply docs/Computer Vision/Object Detection/R-CNN.md:17]: 넵! 위의 답변 내용들을 포함하여 업데이트 하겠습니다 감사합니다~!

**Comment by Journey99** [comment]
민찬님 고생 많으셨습니다~ 내용 정리를 잘 해주신 것 같아요!!
